R/guides-.R

```diff
@@ -741,6 +741,92 @@ Guides <- ggproto(
   }
 )
 
+# Data accessor -----------------------------------------------------------
+
+#' Extract tick information from guides
+#'
+#' `get_guide_data()` builds a plot and extracts information from guide keys. This
+#' information typically contains positions, values and/or labels, depending
+#' on which aesthetic is queried or guide is used.
+#'
+#' @param plot A `ggplot` or `ggplot_build` object.
+#' @param aesthetic A string that describes a single aesthetic for which to
+#'   extract guide information. For example: `"colour"`, `"size"`, `"x"` or
+#'   `"y.sec"`.
+#' @param i,j An integer giving a row (i) and column (j) number of a facet for
+#'   which to return position guide information.
```

Comment:
How does this play with facets from extension packages that might not adhere to a strict grid of panels?

Reply:
Depends on how the facets handle the layout I suppose. From my own package, `facet_manual()` has `.LEFT`, `.RIGHT`, `.TOP` and `.BOTTOM` instead of `ROW` and `COL`, so it won't work.

``` r
devtools::load_all("~/packages/ggplot2")
#> ℹ Loading ggplot2
library(ggh4x)
#> 
#> Attaching package: 'ggh4x'
#> 
#> The following object is masked from 'package:ggplot2':
#> 
#>     guide_axis_logticks

ggplot(mtcars, aes(mpg, wt)) +
  geom_point() +
  facet_manual(
    ~ cyl, design = matrix(c(1,1,NA,NA,NA,2,2,NA,NA,NA,3,3), nrow = 4)
  )
```

![](https://i.imgur.com/Kzbboi6.png)<!-- -->

``` r
get_guide_data(last_plot(), aesthetic = "x", i = 1, j = 1)
#> NULL
```

<sup>Created on 2023-12-14 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Maybe we should just allow a single index to not force a tabular layout. I know that would require some knowledge about how the facet lays out the panels but this is not something for the standard user anyway

Reply:
basically use `[[` indexing rather than `[`

Reply:
I like that idea, and match `PANEL` instead of `ROW`/`COL`. I think `PANEL` should be used in most facet extensions, I'd think.